crates/cxx-qt-gen/src/parser/property.rs

```diff
@@ -46,6 +180,81 @@ mod tests {
         assert_eq!(property.ty, parse_quote! { T });
     }
 
+    #[test]
+    fn test_parse_flags_read() {
+        let mut input: ItemStruct = parse_quote! {
+            #[qproperty(T, name, read)]
+            struct MyStruct;
+        };
+        let property = ParsedQProperty::parse(input.attrs.remove(0)).unwrap();
+        assert_eq!(property.ident, format_ident!("name"));
+        assert_eq!(property.ty, parse_quote! { T });
+    }
+
+    #[test]
+    fn test_parse_flags_all() {
+        let mut input: ItemStruct = parse_quote! {
+            #[qproperty(T, name, read, write, notify)]
+            struct MyStruct;
+        };
+        let property = ParsedQProperty::parse(input.attrs.remove(0)).unwrap();
+        assert_eq!(property.ident, format_ident!("name"));
+        assert_eq!(property.ty, parse_quote! { T });
+        assert!(property.flags.write.is_some());
+        assert!(property.flags.notify.is_some());
+    }
+
+    #[test]
+    fn test_parse_flags_kw() {
+        let mut input: ItemStruct = parse_quote! {
+            #[qproperty(T, name, read = blah, write, notify = blahblah)]
+            struct MyStruct;
+        };
+        let property = ParsedQProperty::parse(input.attrs.remove(0)).unwrap();
+        assert_eq!(property.ident, format_ident!("name"));
+        assert_eq!(property.ty, parse_quote! { T });
+
+        assert!(property.flags.read.is_some());
+        assert_eq!(property.flags.read.unwrap(), format_ident!("blah"));
+
+        assert!(property.flags.write.is_some());
+
+        assert!(property.flags.notify.is_some());
+        let notify = property.flags.notify.unwrap();
+        assert!(notify.is_some());
+        assert_eq!(notify.unwrap(), format_ident!("blahblah"));
+    }
+
+    #[test]
+    fn test_parse_flags_invalid() {
+        let mut input: ItemStruct = parse_quote! {
+            #[qproperty(T, name, read = blah, a, notify = blahblah)]
+            struct MyStruct;
+        };
+        let property = ParsedQProperty::parse(input.attrs.remove(0));
+        assert!(property.is_err())
+    }
+
+    #[test]
+    fn test_parse_missing_flags() {
```

Comment:
maybe change the test name to test_parse_missing_read_flag just to enforce that we are expecting it to fail as `read` is missing

Reply:
done